Rom.py

```diff
@@ -1359,6 +1398,218 @@ def hint_text(dest, ped_hint=False):
+    rom.write_int16(snes_to_pc(0x1af544), 0x1484)
+    rom.write_int16(snes_to_pc(0x1af54c), 0x5484)
+    rom.write_int16(snes_to_pc(0x1af554), 0x14A2)
+    rom.write_int16(snes_to_pc(0x1af55c), 0x54A2)
+    rom.write_int16(snes_to_pc(0x1af564), 0x14A0)
+    rom.write_int16(snes_to_pc(0x1af56c), 0x54A0)
+    rom.write_int16(snes_to_pc(0x1af574), 0x148E)
+    rom.write_int16(snes_to_pc(0x1af57c), 0x548E)
+    rom.write_int16(snes_to_pc(0x1af584), 0x14AE)
+    rom.write_int16(snes_to_pc(0x1af58c), 0x54AE)
+    rom.write_byte(snes_to_pc(0x00DB9D), 0x1A)  # castle hole graphics
+    rom.write_byte(snes_to_pc(0x00DC09), 0x1A)
+    rom.write_byte(snes_to_pc(0x00D009), 0x31)
+    rom.write_byte(snes_to_pc(0x00D0e8), 0xE0)
+    rom.write_byte(snes_to_pc(0x00D1c7), 0x00)
+    rom.write_int16(snes_to_pc(0x1BE8DA), 0x39AD)
+    rom.write_byte(0xF6E58, 0x80)  # no whirlpool under castle gate
+    rom.write_bytes(0x0086E, [0x5C, 0x00, 0xA0, 0xA1])  # TR tail
+    rom.write_bytes(snes_to_pc(0x1BC67A), [0x2E, 0x0B, 0x82])  # add warps under rocks
+    rom.write_bytes(snes_to_pc(0x1BC81E), [0x94, 0x1D, 0x82])
+    rom.write_bytes(snes_to_pc(0x1BC655), [0x4A, 0x1D, 0x82])
+    rom.write_bytes(snes_to_pc(0x1BC80D), [0xB2, 0x0B, 0x82])
+    rom.write_bytes(snes_to_pc(0x1BC3DF), [0xD8, 0xD1])
+    rom.write_bytes(snes_to_pc(0x1BD1D8), [0xA8, 0x02, 0x82, 0xFF, 0xFF])
+    rom.write_bytes(snes_to_pc(0x1BC85A), [0x50, 0x0F, 0x82])
+    rom.write_int16(0xDB96F + 2 * 0x35, 0x001B)  # move pyramid exit door
+    rom.write_int16(0xDBA71 + 2 * 0x35, 0x06A4)
+    if world.shuffle == 'vanilla':
+        rom.write_byte(0xDBB73 + 0x35, 0x36)
+    rom.write_byte(snes_to_pc(0x09D436), 0xF3)  # remove castle gate warp
+    if world.shuffle == 'vanilla':
+        rom.write_int16(0x15AEE + 2 * 0x37, 0x0010)  # pyramid exit to new hc area
+        rom.write_byte(0x15B8C + 0x37, 0x1B)
+        rom.write_int16(0x15BDB + 2 * 0x37, 0x0418)
+        rom.write_int16(0x15C79 + 2 * 0x37, 0x0679)
+        rom.write_int16(0x15D17 + 2 * 0x37, 0x06B4)
+        rom.write_int16(0x15DB5 + 2 * 0x37, 0x06C6)
+        rom.write_int16(0x15E53 + 2 * 0x37, 0x0738)
+        rom.write_int16(0x15EF1 + 2 * 0x37, 0x06E6)
+        rom.write_int16(0x15F8F + 2 * 0x37, 0x0733)
+        rom.write_byte(0x1602D + 0x37, 0x07)
+        rom.write_byte(0x1607C + 0x37, 0xF9)
+        rom.write_int16(0x160CB + 2 * 0x37, 0x0000)
+        rom.write_int16(0x16169 + 2 * 0x37, 0x0000)
+    rom.write_bytes(snes_to_pc(0x1BC387), [0xDD, 0xD1])
+    rom.write_bytes(snes_to_pc(0x1BD1DD), [0xA4, 0x06, 0x82, 0x9E, 0x06, 0x82, 0xFF, 0xFF])
+    rom.write_byte(0x180089, 0x01)  # open TR after exit
+    rom.write_byte(snes_to_pc(0x0ABFBB), 0x90)
+    rom.write_byte(snes_to_pc(0x0280A6), 0xD0)
+    rom.write_bytes(snes_to_pc(0x06B2AB), [0xF0, 0xE1, 0x05])
+
+def patch_shuffled_dark_sanc(world, rom, player):
+    dark_sanc_entrance = re.split('\(Pl', str(world.get_region('Inverted Dark Sanctuary', player).entrances[0]))[0].rstrip()
```

Comment:
Question: Is there a reason you can just get the playerless name off the entrance? I'm pretty sure it has some property or other with name and without the player number information. That would seem just slightly cleaner.

Reply:
Yeah, the `Entrance.name` attribute seems to work